File: server/data/results/translations/anchored/java/two_sum.java
import java.util.HashMap;
import java.util.Map;

public class Main {

    public static int[] two_sum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (seen.containsKey(complement)) {
                return new int[]{seen.get(complement), i};
            } else {
                seen.put(nums[i], i);
            }
        }
        return new int[]{-1, -1};
    }

    static class TestCase {
        int[] nums;
        int target;

        TestCase(int[] nums, int target) {
            this.nums = nums;
            this.target = target;
        }
    }

    public static void main(String[] args) {
        TestCase[] cases = new TestCase[]{
            new TestCase(new int[]{2, 7, 11, 15}, 9),
            new TestCase(new int[]{3, 2, 4}, 6),
            new TestCase(new int[]{1, 2, 3}, 100)
        };

        for (TestCase c : cases) {
            int[] pair = two_sum(c.nums, c.target);
            System.out.println(pair[0] + " " + pair[1]);
        }
    }
}

File: server/data/results/translations/anchored/cpp/two_sum.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <utility>

std::vector<int> two_sum(const std::vector<int>& nums, int target) {
    std::unordered_map<int, int> seen;
    for (int i = 0; i < static_cast<int>(nums.size()); ++i) {
        int complement = target - nums[i];
        if (seen.find(complement) != seen.end()) {
            return {seen[complement], i};
        } else {
            seen[nums[i]] = i;
        }
    }
    return {-1, -1};
}

int main() {
    std::vector<std::pair<std::vector<int>, int>> cases = {
        {{2, 7, 11, 15}, 9},
        {{3, 2, 4}, 6},
        {{1, 2, 3}, 100}
    };

    for (const auto& item : cases) {
        const auto& nums = item.first;
        int target = item.second;
        std::vector<int> pair = two_sum(nums, target);
        std::cout << std::to_string(pair[0]) + " " + std::to_string(pair[1]) << std::endl;
    }

    return 0;
}